Handle ignore back press

DiogoTavares87 opened the pull request:
* Implemented logic to ignore back press clicks when the dialog is not
cancellable.

DiogoTavares87 committed:
NO_JIRA Handle ignore back press

* Implemented logic to ignore back press clicks when the dialog is not
cancellable.

neteinstein commented:
@bruno-garces @pedro-belchior @DiogoTavares87 @Jmvcaseiro 

So... when a dialog is not cancellable, it shouldn't allow the backpress, anybody against it?

If so, we would delete this new tag, and just have cancellable as a generic one.

bruno-garces reviewed (commented):
agree @neteinstein

DiogoTavares87 committed:
[NO_JIRA] Handle ignore back press

* Reverted previous logic and removed the handling of back press,
so from now on, the back will be ignored if the dialog is set to
cancellable = false.

neteinstein commented:
```
com.mindera.skeletoid.webview.cookies.CookiesUtilsTest > testGetCookiesFromUrlNoCookies FAILED
    java.lang.AssertionError
com.mindera.skeletoid.webview.cookies.CookiesUtilsTest > testGetCookieNoCookies FAILED
    java.lang.AssertionError
```

The build is failing with this. It's not related with this PR. @pedro-belchior didn't you fix this before?

DiogoTavares87 committed:
[NO_JIRA] Handle ignore back press

* Removed unused BACK_PRESSED DialogState

neteinstein committed:
[fix] Fixed cookies unit tests

neteinstein committed:
[fix] Delete old unused var

neteinstein committed:
[fix] Keyboard hide: unit test
